src/hooks/useNetworkSwitch.ts

```diff
@@ -17,25 +18,50 @@ export type UseNetworkSwitchProps = {
 export const useNetworkSwitch = ({ onSelectNetworkCallback }: UseNetworkSwitchProps = {}) => {
   const { connector, chainId, account } = useActiveWeb3React()
   const unsupportedChainIdError = useUnsupportedChainIdError()
+  const navigate = useNavigate()
+  const { pathname } = useLocation()
 
   const selectNetwork = useCallback(
-    (optionChainId?: ChainId) => {
+    async (optionChainId?: ChainId) => {
       if (optionChainId === undefined || (optionChainId === chainId && !unsupportedChainIdError)) return
+
+      let changeChainId: Promise<unknown> | undefined
       if (!account && !unsupportedChainIdError && connector instanceof CustomNetworkConnector)
         connector.changeChainId(optionChainId)
```

Comment:
Just so I can understand. Why aren't we checking this promise?

Reply:
Because it's not a promise. Inside that function there is a sync call... unlike the other methods.

Reply:
Got it. They are from different classes. `CustomWalletLinkConnector` returns a promise, while `CustomNetworkConnector` doesn't!

Reply:
Small correction for clarity.
That specific method of `CustomNetworkConnector` doesn't return a promise. That class has other methods that do return promises.
But you got the gist of it.